packages/@jsii/go-runtime/jsii-calc-test/main_test.go

```diff
@@ -14,64 +18,137 @@ func TestMain(m *testing.M) {
 	os.Exit(status)
 }
 
+func initCalculator() calc.CalculatorIface {
+	calculatorProps := calc.CalculatorProps{InitialValue: float64(10), MaximumValue: math.MaxFloat64}
+	return calc.NewCalculator(&calculatorProps)
+}
+
 func TestCalculator(t *testing.T) {
-	var initialValue float64 = 10
-	calculatorProps := calc.CalculatorProps{InitialValue: initialValue, MaximumValue: math.MaxFloat64}
-	calculator := calc.NewCalculator(&calculatorProps)
+	// Object creation
+	t.Run("Object creation", func(t *testing.T) {
+		calculator := initCalculator()
+		if reflect.ValueOf(calculator).IsZero() {
+			t.Errorf("Expected calculator object to be valid")
+		}
+	})
 
-	t.Run("Primitive Property Access", func(t *testing.T) {
+	t.Run("Property access", func(t *testing.T) {
+		calculator := initCalculator()
+		expected := float64(10)
 		actual := calculator.GetValue()
-		if actual != initialValue {
-			t.Errorf("Expected: %f; Actual %f;", initialValue, actual)
+		if actual != expected {
+			t.Errorf("Expected: %f; Actual %f;", expected, actual)
 		}
 	})
 
-	t.Run("Method Call Effect", func(t *testing.T) {
-		var factor float64 = 3
-		calculator.Mul(factor)
-		var expected float64 = initialValue * factor
+	t.Run("Property mutation", func(t *testing.T) {
+		calculator := initCalculator()
+		var newVal float64 = 12345
+		currentProps := calclib.NewNumber(newVal)
+		calculator.SetCurr(currentProps)
 		actual := calculator.GetValue()
+		if newVal != actual {
+			t.Errorf("Expected: %f; Actual %f;", newVal, actual)
+		}
+	})
 
-		if actual != expected {
-			t.Errorf("Expected: %f; Actual %f;", expected, actual)
+	t.Run("Method with side effect", func(t *testing.T) {
+		initial, factor := float64(10), float64(3)
+		calculator := calc.NewCalculator(&calc.CalculatorProps{InitialValue: initial, MaximumValue: math.MaxFloat64})
+		calculator.Mul(factor)
+		expectedProduct := initial * factor
+		actualProduct := calculator.GetValue()
+		if actualProduct != expectedProduct {
+			t.Errorf("Expected quotient: %f; Actual %f;", expectedProduct, actualProduct)
 		}
 	})
 
-	t.Run("Method Call Returns Any Value", func(t *testing.T) {
+	t.Run("Method with interface{} return type", func(t *testing.T) {
+		calculator := initCalculator()
+		expectedTypeName := "Calculator"
+		actualTypeName := calculator.TypeName()
 		// JSII tells us this return value is an "any" type. Therefore the
 		// value received by go is type `interface{}` and can be further
 		// specialized using reflection.
-		expected := "Calculator"
-		actual := calculator.TypeName()
-		switch retType := actual.(type) {
+		switch retType := actualTypeName.(type) {
```

Comment:
Yeah these are funky because as far as I can tell its the only way to do a [type matching](https://tour.golang.org/methods/16) without reflection. It seems to be the preferred way of handling this.